src/features/ProfilePage/validators.test.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  validateBirthYear,
+  validateDisplayNameLength,
+  validateEmail,
+  validatePasswordLength,
+} from './validators';
+
+describe('Validation Functions', () => {
+  describe('validateBirthYear', () => {
+    it('should return false for a birth year less than MIN_BIRTH_YEAR', () => {
+      expect(validateBirthYear(1899)).toBe(false);
+    });
+
+    it('should return false for a birth year greater than MAX_BIRTH_YEAR', () => {
+      const futureBirthYear = new Date().getFullYear() - 16; // Younger than 17
+      expect(validateBirthYear(futureBirthYear)).toBe(false);
+    });
+
+    it('should return true for a valid birth year within range', () => {
+      expect(validateBirthYear(1990)).toBe(true);
+    });
+  });
+
+  describe('validateDisplayNameLength', () => {
+    it('should return false if display name length is less than DISPLAY_NAME_MIN_LENGTH', () => {
+      expect(validateDisplayNameLength('A')).toBe(false);
+    });
+
+    it('should return true if display name length is greater than or equal to DISPLAY_NAME_MIN_LENGTH', () => {
+      expect(validateDisplayNameLength('AB')).toBe(true);
+    });
+  });
+
+  describe('validateEmail', () => {
+    it('should return true for an empty email (optional field)', () => {
+      expect(validateEmail('')).toBe(true);
+    });
+
+    it('should return true for a valid email', () => {
+      expect(validateEmail('test@example.com')).toBeTruthy();
```

Comment:
We expect the result to be "truthy"? 

The signature is a bit loose imo
```typescript
const validateEmail: (email: string) => true | RegExpMatchArray | null
```

Could we just return boolean?

Btw the ylitse-app repo has great test for this already. 
https://github.com/sos-lapsikyla/ylitse-app/blob/master/src/lib/__tests__/validators.ts#L21

Since we have it existing, I would just use it here too. Easy copy-paste with a lot of test-cases

Reply:
Yes, I'll fix that! I also see from the ylitse-app repo that there are more validation done in the backend. I updated this backlog item: https://trello.com/c/i9mE16Ry/986-ui-password-validation

Reply:
I now copied the exact test for the email validator from ylitse-app. I had to change the regex a bit and add a maximum email length to get the test cases to pass. So now the validator is identical to the ylitse-app one.